File: truthiness.rb
def test(thing)
  if thing
    puts "it's true!"
  else
    puts "it's false!"
  end
end

test('') # => it's true!
test(nil) # => it's false!
test(true) # => it's true!
test(false) # => it's false!
test(['']) # => it's true!
test([true]) # => it's true!
test([false]) # => it's true!
test([]) # => it's true!
test(0) # => it's true!
test(1) # => it's true!

# summary: nil and false are falsey
# not unreasonable but kind of annoying, you still need to check
# false specifically to be strict, but you also need to compare
# a bunch of presence or numerical things if you want "falsiness"

File: truthiness.py
def test(thing):
    if thing:
        print("it's True!")
    else:
        print("it's False!")

test('') # => it's False!
test(None) # => it's False!
test(True) # => it's True!
test(False) # => it's False!
test(['']) # => it's True!
test([True]) # => it's True!
test([False]) # => it's True!
test([]) # => it's False!
test(0) # => it's False
test(1) # => it's True!

# summary: empty data structures, None, False, and 0 are falsey
# a bit loose, but more useful (just check False specifically if you want to be strict)
